cdap-app-fabric/src/main/java/io/cdap/cdap/app/store/preview/PreviewStore.java

```diff
@@ -91,4 +92,30 @@
    */
   @Nullable
   PreviewStatus getPreviewStatus(ApplicationId applicationId);
+
+  /**
+   * Adds the preview request for given application id in waiting state.
+   * @param applicationId the application id corresponding to the request
+   * @param appRequest preview request configuration
+   */
+  void addToWaitingState(ApplicationId applicationId, AppRequest appRequest);
```

Comment:
Why need a specific method for the waiting state? Shouldn't be the same as other state transition that uses the `setPreviewStatus` method?

Reply:
Renamed this method to `add`. This method also sets the status as WAITING since thats the initial status of the preview.